superset/__init__.py

```diff
@@ -160,4 +162,16 @@ def index(self):
 module_datasource_map.update(app.config.get('ADDITIONAL_MODULE_DS_MAP'))
 ConnectorRegistry.register_sources(module_datasource_map)
 
+allowed_certs = conf.get('ALLOWED_CERT_COMMON_NAMES')
+if allowed_certs:
+    @app.before_request
+    def is_valid_request():
+        cert = request.headers.get('X-CLIENT-SSL-CERT')
```

Comment:
Is this something specific to `internal_auth` or a certain flavor of proxies? Google "X-CLIENT-SSL-CERT" doesn't give out much result so that doesn't seem like a standard.